src/main/kotlin/org/jetbrains/research/ml/codetracker/Plugin.kt

```diff
@@ -1,21 +1,107 @@
 package org.jetbrains.research.ml.codetracker
 
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
 import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.application.ex.ApplicationEx
 import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import net.lingala.zip4j.ZipFile
+import java.io.File
+import java.io.InputStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+
 
 enum class TestMode {
     ON,
     OFF
 }
 
+/**
+ * Represents a plugin required by Codetracker
+ */
+data class RequiredPlugin(val name: String, val zipFile: String) {
+    private val resourceFolder = "requiredplugins"
+    private val logger: Logger = Logger.getInstance(javaClass)
```

Comment:
I think it would be better to make its companion object. Do we need to have a new instance of the logger and the resource Folder each time when a new Required Plugin object is created?